### ISSUE TITLE
Handle Cross Origin XHR for Packaged Applications

### ISSUE THREAD

**Issue by mounirlamouri** [sysapps/runtime#22]
Summary of the discussion in the mailing-list:
http://www.w3.org/wiki/System_Applications_WG:_Cross_Origin_XHR

**Comment by marcoscaceres** [comment]
Nice summary. I'm for not changing default behaviour of XHR, but allowing faking the origin on privileged applications. I described a potential solution here:

http://lists.w3.org/Archives/Public/public-sysapps/2013Mar/0183.html

**Comment by mounirlamouri** [comment]
Just to make sure credit is given, the summary has been made by John Lyle, not me.

I think I would prefer a solution along the lines of the one you proposed which is very close to what @sicking proposed too.

**Comment by marcoscaceres** [comment]
I wonder if we should discuss implications with @annevk also? As Editor of both XHR and CORS, he probably has some helpful views on this.

**Comment by annevk** [comment]
This does not affect just XMLHttpRequest. This would also affect &lt;img crossorigin>, etc. If you want to introduce a new security model, you have to think it through, and not patch APIs here and there.

**Comment by marcoscaceres** [comment]
I agree with @annevk and [outlined a similar argument on the list](http://lists.w3.org/Archives/Public/public-sysapps/2013Apr/0033.html).
